### ISSUE TITLE
Prompt.file filter does not provide enough context to filter files

### ISSUE THREAD

**Issue by danielo515** [Effect-TS/effect#3393]
### What is the problem this feature would solve?

Hello. I'm facing some problems using the Prompt.file utility.
The first surprising thing is that using a filter to check the file extension can also remove the ability to travel to upper directories, so instead of just checking for the extension I also have to account for `..`, not very annoying, but worth mentioning IMO.

Second is that the provided path is not absolute, neither relative to the starting directory where the execution started, so if you want to check if the file is a directory (for example) and the user has navigated to a subfolder or to a parent folder, you have no way to know where is the actual file your filer has to check.

Here is an example of what I'm trying:

```ts
  const xmlPath = yield* Prompt.file({
    filter: (p) =>
      Effect.gen(function* (_) {
        console.log(p);
        const stat = yield* fs.stat(path.resolve(p)).pipe(Effect.orDie);
        return stat.type === "Directory" || p.endsWith(".xml");
      }),
    message: "Please select one XML file",
  });
```

### What is the feature you are proposing to solve the problem?

The filter should receive a full path, or a relative path, or a file handle

### What alternatives have you considered?

The only one I can think about is to implement my own custom input

**Comment by fubhy** [comment]
Closing this as a stale enhancement request. The request is understandable, but there has not been recent interest or maintainer activity to keep it actionable.

**Comment by danielo515** [comment]
I don't think this should be closed. If the request is understandable, and the only problem is that nobody find time to implement it, I don't think closing it will help anyone.
And closing it as completed is misleading

**Comment by fubhy** [comment]
Feel free to re-open and submit a pull request once we've merged the v4 codebase into this repository (imminent).

**Comment by fubhy** [comment]
> I don't think this should be closed. If the request is understandable, and the only problem is that nobody find time to implement it, I don't think closing it will help anyone. And closing it as completed is misleading

Also, tbh. "nobody has found time to implement it" is actually a pretty damn good measure for whether something should be considered relevant.

Reacting to the closure by reopening is a valid way to express your continued interest and helps us filter through the large pile of issues. So please don't take offense if I go around and close issues broadly. Simply reopening the ones that still matter to you is perfectly fine.

**Comment by tim-smart** [comment]
Automated message: Effect v3 is now under a feature freeze and is now only considering bug fixes.

If this issue is still relevant for v4, then feel free to reopen :)
